test/integration/private_connection/controls/forseti.rb

```diff
@@ -0,0 +1,180 @@
+
+    # Enumerate the files that we expect to be present. This fixture ensures that we
+    # don't silently drop a rules file.
+    let(:expected_files) do
+      %w[
+        rules/audit_logging_rules.yaml
+        rules/bigquery_rules.yaml
+        rules/blacklist_rules.yaml
+        rules/bucket_rules.yaml
+        rules/cloudsql_rules.yaml
+        rules/enabled_apis_rules.yaml
+        rules/external_project_access_rules.yaml
+        rules/firewall_rules.yaml
+        rules/forwarding_rules.yaml
+        rules/group_rules.yaml
+        rules/groups_settings_rules.yaml
+        rules/iam_rules.yaml
+        rules/iap_rules.yaml
+        rules/instance_network_interface_rules.yaml
+        rules/ke_rules.yaml
+        rules/ke_scanner_rules.yaml
+        rules/lien_rules.yaml
+        rules/location_rules.yaml
+        rules/log_sink_rules.yaml
+        rules/resource_rules.yaml
+        rules/retention_rules.yaml
+        rules/role_rules.yaml
+        rules/service_account_key_rules.yaml
+      ]
+    end
+
+    # Enumerate the files that are present in the rules directory  This fixture ensures
+    # that we don't miss an included rules file.
+    let(:present_files) do
+      template_dir = File.expand_path(
+        "../../../../modules/rules/templates/rules",
+        __dir__
+      )
+      Dir.glob("#{template_dir}/*.yaml").map {|file| "rules/#{File.basename(file)}" }
+    end
+
+    let(:files) { expected_files | present_files }
+
+    its('object_names') { should include(*files) }
+  end
+
+  describe google_service_account(project: project_id, name: forseti_server_service_account) do
+    its(:display_name) { should eq "Forseti Server Service Account" }
+  end
+
+  describe google_compute_firewall(project: project_id, name: "forseti-server-allow-grpc-#{suffix}") do
+    # let(:allowed) { subject.allowed.map(&:item) }
+
+    its('source_ranges') { should eq ["10.128.0.0/9"] }
+    its('direction') { should eq 'INGRESS' }
+    its('priority') { should eq 100 }
+
+    # replace commented lines with documented way to validate port/protocol
+    it { should allow_port_protocol("50051", "tcp") }
+    it { should allow_port_protocol("50052", "tcp") }
+
+    # it "allows gRPC traffic" do
+    #   expect(allowed).to contain_exactly({ip_protocol: "tcp", ports: ["50051", "50052"]})
+    # end
+  end
+
+  describe google_compute_firewall(project: project_id, name: "forseti-server-deny-all-#{suffix}") do
+    # let(:denied) { subject.denied.map(&:item) }
+
+    its('source_ranges') { should eq ["0.0.0.0/0"] }
+    its('direction') { should eq 'INGRESS' }
+    its('priority') { should eq 200 }
+
+    # TODO: uncomment/fix once the resource will have a method to verify the protocol is allowed/denied for all ports
+    # it "denies TCP, UDP, and ICMP" do
+    #   expect(denied).to contain_exactly(
+    #     {ip_protocol: "icmp"},
+    #     {ip_protocol: "tcp"},
+    #     {ip_protocol: "udp"}
+    #   )
+    # end
```

Comment:
there is no support for matching a rule's protocol without port or for all ports. the lines are commented until a Ruby's programmer will add a library function or the official support won't be added